.github/workflows/native-image.yml

```diff
@@ -0,0 +1,50 @@
+name: Native Image
+
+on:
+  push:
+    branches:
+      - native-image
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  workflow_config:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+
+      - name: Get jar
+        run: |
+          wget "https://github.com/plantuml/plantuml/releases/download/${PLANTUML_VERSION}/plantuml-${PLANTUML_VERSION#?}.jar" -O plantuml.jar
+        env:
+          PLANTUML_VERSION: "v1.2023.4"
+
+      - name: Install system requirements
+        run: sudo apt install build-essential libz-dev zlib1g-dev
+
+      - name: Get GraalVM
+        run: |
+          wget "https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-${GRAALVM_VERSION}/graalvm-ce-java17-linux-amd64-${GRAALVM_VERSION}.tar.gz" -O graalvm-ce.tar.gz
+          mkdir graalvm-ce
+          tar xvzf graalvm-ce.tar.gz -C graalvm-ce --strip-components 1
+          ./graalvm-ce/bin/gu install native-image
+        env:
+          GRAALVM_VERSION: "22.3.1"
+
+      - name: Generate GraalVM configuration
+        run: |
+          mkdir native-image-config-dir
+          echo 'Bob->Alice: Hello' | ./graalvm-ce/bin/java -agentlib:native-image-agent=config-output-dir=./native-image-config-dir -jar plantuml.jar -tpng -pipe > out.png
```

Comment:
Hi @ggrossetie, and all,

_Sorry for the newbie question..._

What are the meaning of those line?
Is an example as `Bob->Alice: Hello` is mandatory in order to generate the `GraalVM configuration`?

Regards.

Reply:
This process is called "Assisted Configuration with Tracing Agent": https://www.graalvm.org/22.0/reference-manual/native-image/Agent/

> Is an example as Bob->Alice: Hello is mandatory in order to generate the GraalVM configuration?

It's just an example and we might need to run PlantUML more than once to trigger all execution paths: "It can be necessary to run the target application more than once with different inputs to trigger separate execution paths for a better coverage of dynamic accesses. The agent supports this with the config-merge-dir option which adds the intercepted accesses to an existing set of configuration files"

For now it seems sufficient to use a simple example such as "Bob->Alice: Hello".